tests/std/tests/GH_000431_equal_family/test.cpp

```diff
@@ -244,7 +244,11 @@ void test_algorithms(EqualFn equal_fn) {
         assert(equal_fn(begin(arr2), end(arr2), begin(arr3), end(arr3), equal_to<short>{}));
         assert(equal_fn(begin(arr2), end(arr2), begin(arr3), end(arr3), equal_to<int>{}));
 
+#if !_HAS_CXX23
+        // The return type of equal_to<void>::operator()(T, U) SFINAEs
+        // when T and U are different enumeration types.
         assert(equal_fn(begin(arr3), end(arr3), begin(arr4), end(arr4), equal_to<>{}));
+#endif // !_HAS_CXX23

```

Comment:
Built-in comparison between different enumeration types is removed in C++26 via WG21-P2864R2.

It seems that Clang only applies the removal to C++26 mode ([Godbolt link](https://godbolt.org/z/bhf5zjx1M)). Why is C++23 mode affected?

Reply:
Possibly because we're passing `/std:c++latest` and it's being detected as C++26?

Reply:
Yes, `/std:c++latest` is probably mapped to `-std=c++2c` by Clang 18+ so C++26 core features become available ([Godbolt link](https://godbolt.org/z/vr9TrrM8e)). 😿 There seems an issue that we can't specify C++23 mode because `/std:c++23` is not yet available.

Reply:
Really this ought to be `// TRANSITION, _HAS_CXX26` but it's not worth resetting testing.